config/config_test.go

```diff
@@ -0,0 +1,333 @@
+					Hosts: "consul.dc1.address",
+				},
+				ConsulLogin: ConsulLogin{
+					Enabled:    true,
+					Method:     "test-auth-method",
+					Datacenter: "test-dc",
+					Meta:       map[string]string{"key1": "value1", "key2": "value2"},
+				},
+			},
+			taskMeta: awsutil.ECSTaskMeta{
+				Cluster: "test-cluster",
+				TaskARN: "arn:aws:ecs:us-east-1:123456789:task/test/abcdef",
+				Family:  "family-service",
+			},
+			expConfig: func(t awsutil.ECSTaskMeta) discovery.Config {
+				return discovery.Config{
+					Addresses: "consul.dc1.address",
+					Credentials: discovery.Credentials{
+						Type: discovery.CredentialsTypeLogin,
+						Login: discovery.LoginCredential{
+							AuthMethod: "test-auth-method",
+							Datacenter: "test-dc",
+							Meta: map[string]string{
+								"key1":                         "value1",
+								"key2":                         "value2",
+								"consul.hashicorp.com/task-id": t.TaskID(),
+								"consul.hashicorp.com/cluster": t.Cluster,
+							},
+						},
+					},
+				}
+			},
+		},
+	}
+
+	for name, c := range cases {
+		t.Run(name, func(t *testing.T) {
+			var srvConfig testutil.ServerConfigCallback
+			if c.cfg.ConsulLogin.Enabled {
+				if testutil.EnterpriseFlag() {
+					c.cfg.Service.Namespace = "test-namespace"
+					c.cfg.Service.Partition = "test-partition"
+				}
+
+				// Enable ACLs to test with the auth method
+				srvConfig = testutil.ConsulACLConfigFn
+				_, apiCfg := testutil.ConsulServer(t, srvConfig)
+				consulClient, err := api.NewClient(apiCfg)
+				require.NoError(t, err)
+				fakeAws := testutil.AuthMethodInit(t, consulClient, c.taskMeta.Family, c.cfg.ConsulLogin.Method)
+
+				// Use the fake local AWS server.
+				c.cfg.ConsulLogin.STSEndpoint = fakeAws.URL + "/sts"
+			}
+
+			cfg, err := c.cfg.ConsulServerConnMgrConfig(c.taskMeta)
+			require.NoError(t, err)
+
+			expectedCfg := c.expConfig(c.taskMeta)
+			if testutil.EnterpriseFlag() {
+				expectedCfg.Credentials.Login.Namespace = "test-namespace"
+				expectedCfg.Credentials.Login.Partition = "test-partition"
+			}
+
+			require.Equal(t, expectedCfg.Addresses, cfg.Addresses)
+
+			if c.cfg.ConsulLogin.Enabled {
+				require.Equal(t, expectedCfg.Credentials.Type, cfg.Credentials.Type)
+				require.Equal(t, expectedCfg.Credentials.Login.AuthMethod, cfg.Credentials.Login.AuthMethod)
+				require.Equal(t, expectedCfg.Credentials.Login.Namespace, cfg.Credentials.Login.Namespace)
+				require.Equal(t, expectedCfg.Credentials.Login.Partition, cfg.Credentials.Login.Partition)
+				require.Equal(t, expectedCfg.Credentials.Login.Meta, cfg.Credentials.Login.Meta)
+				require.Equal(t, expectedCfg.Credentials.Login.Datacenter, cfg.Credentials.Login.Datacenter)
+				require.NotEmpty(t, cfg.Credentials.Login.BearerToken)
+			}
+		})
+	}
+}
+
+func TestConsulServerConnManagerConfig_TLS(t *testing.T) {
+	caFile, err := os.CreateTemp("", "")
+	t.Cleanup(func() {
+		_ = os.RemoveAll(caFile.Name())
+	})
+	require.NoError(t, err)
```

Comment:
nit: check for error before adding the cleanup

```suggestion
	require.NoError(t, err)
	t.Cleanup(func() {
		_ = os.RemoveAll(caFile.Name())
	})
```